Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs

```diff
@@ -1103,10 +980,135 @@ void HandleChildPropertyChanged(object sender, PropertyChangedEventArgs e)
 					UpdateLargeTitles();
 				else if (e.PropertyName == NavigationPage.TitleIconProperty.PropertyName ||
 					 e.PropertyName == NavigationPage.TitleViewProperty.PropertyName)
+					UpdateTitleArea(Child);
+			}
+
+
+			internal void UpdateLeftBarButtonItem(Page pageBeingRemoved = null)
+			{
+				NavigationRenderer n;
+				if (!_navigation.TryGetTarget(out n))
+					return;
+
+				var currentChild = this.Child;
+				var firstPage = n.NavPage.Pages.FirstOrDefault();
+
+
+				if (n._parentMasterDetailPage == null)
+					return;
+
+				if (firstPage != pageBeingRemoved && currentChild != firstPage && NavigationPage.GetHasBackButton(currentChild))
+				{
+					NavigationItem.LeftBarButtonItem = null;
```

Comment:
Actual behavior changes here
When telling the back button to start appearing again the MDP title wouldn't go away.  Once you set ShowBackButton to false you were just stuck with never being able to make it show up again.